File: matlab/tools/filters/notch_filter.m
function values = notch_filter(values,fs,varargin)

notch_freq = 60;
if nargin == 3
    notch_freq = varargin{1};
end

f = designfilt('bandstopiir','FilterOrder',4, ...
   'HalfPowerFrequency1',notch_freq-1,'HalfPowerFrequency2',notch_freq+1, ...
   'DesignMethod','butter','SampleRate',fs);
%fvtool(f)
for i = 1:size(values,2) 
    
    eeg = values(:,i);
    
    if sum(~isnan(eeg)) == 0
        continue
    end
    
    eeg(isnan(eeg)) = mean(eeg,'omitnan');
    eeg = filtfilt(f,eeg);   
    values(:,i) = eeg;
end


end

File: python/tools/notch_filter.py
from scipy.signal import iirnotch, filtfilt, butter



def notch_filter(data,fs,notch_freq = 60):

    b, a = iirnotch(notch_freq, notch_freq/2, fs=fs)
    # b, a = butter(4, [59,61], 'bandstop', fs) 

    y = filtfilt(b, a, data, axis = 0)

    return y

